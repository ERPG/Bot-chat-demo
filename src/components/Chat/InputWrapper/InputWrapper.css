.landbot-input-container {
  position: relative;
  background-color: var(--dark-blue-2);
  border-bottom-left-radius: var(--chat-border-radius);
  border-bottom-right-radius: var(--chat-border-radius);
}