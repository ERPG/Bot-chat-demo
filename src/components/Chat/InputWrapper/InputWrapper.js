import React, { useContext, useState } from 'react';
import { GlobalLandbotContext } from '../../../contexts/LandbotContext';
import Button from '../../common/Button/Button';
import Input from '../../common/Input/Input';

import './InputWrapper.css';

/** The InputWrapper is the container that have the input and the action button to send messages */

const InputWrapper = () => {
  const [inputValue, setInputValue] = useState('');

  const Core = useContext(GlobalLandbotContext);

  const onChangeInputHandler = (e) =>{
    setInputValue(e.target.value)
  }

  const onKeyUpHandler = (e) =>{
    if (e.key === 'Enter') {
      e.preventDefault();
      submit();
    }
  }

  const submit = () => {
    if (inputValue !== '') {
      Core.sendMessage({ message: inputValue });
      setInputValue('');
    }
  };

  return (
    <div className="landbot-input-container" data-landbot="input-container">
      <div className="field">
        <div className="control">
          <Input
            placeholder={"Type here..."}
            value={inputValue}
            onChange={onChangeInputHandler}
            onKeyUp={onKeyUpHandler}
          />
          <Button onClick={submit} isDisabled={inputValue === ''}>
            <span className="icon is-large">
              <i className="fas fa-paper-plane fa-lg"></i>
            </span>
          </Button>
        </div>
      </div>
    </div>
  )
}

export default InputWrapper;