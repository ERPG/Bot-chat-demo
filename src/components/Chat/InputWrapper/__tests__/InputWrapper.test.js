import React from 'react';
import { render } from '@testing-library/react';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';
import InputWrapper from '../InputWrapper';

describe('InputWrapper component', () => {
  const inputContainerSelector = 'input-container';

  it('Should render component without errors', () => {
    const { container } = render(<InputWrapper />);
    expect(container).toMatchSnapshot();
  });

  it('Should have a container', () => {
    const { getByDataLb } = customRender(<InputWrapper />);
    const element = getByDataLb(inputContainerSelector)
    expect(element).toBeDefined();
  });
});