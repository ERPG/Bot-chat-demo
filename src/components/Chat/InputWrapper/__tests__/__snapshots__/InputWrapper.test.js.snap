// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWrapper component Should render component without errors 1`] = `
<div>
  <div
    class="landbot-input-container"
    data-landbot="input-container"
  >
    <div
      class="field"
    >
      <div
        class="control"
      >
        <input
          class="landbot-input"
          data-landbot="chat-input"
          placeholder="Type here..."
          type="text"
          value=""
        />
        <button
          class="button landbot-input-send"
          data-landbot="send-button"
          disabled=""
        >
          <span
            class="icon is-large"
          >
            <i
              class="fas fa-paper-plane fa-lg"
            />
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
